docs/user_guide/modeling_extensions/uncertainty_propagation/index.rst

```diff
@@ -0,0 +1,132 @@
+
+Based on a first-order error propagation formula, the variance of the objective function :math:`Var[f(x^*,p^*)]` and constraints :math:`c(x^*,p^*)` are: 
+
+.. math::
+
+    Var[f(x^*,p^*)] = \frac{\partial f}{\partial p}  \Sigma_p  \frac{\partial f}{\partial p}^T + \frac{\partial f}{\partial x}\frac{\partial x}{\partial p} \Sigma_p   \frac{\partial x}{\partial p}^T\frac{\partial f}{\partial x}^T, 
+
+    Var[c_1(x^*,p^*)] = \frac{\partial c_1}{\partial p}  \Sigma_p  \frac{\partial c_1}{\partial p}^T + \frac{\partial c_1}{\partial x}\frac{\partial x}{\partial p} \Sigma_p   \frac{\partial x}{\partial p}^T\frac{\partial c_1}{\partial x}^T, 
+
+    \ \ \ \ \ \ \ \ \ \  \ \ \ \ \ \ \ \ \ \ \ \ \ \ \  \ \ \ \ \  \ \ \ \ \ \ \ \ \ \  \ \ \ \ \ \vdots
+
+    Var[c_k(x^*,p^*)] = \frac{\partial c_k}{\partial p}  \Sigma_p  \frac{\partial c_k}{\partial p}^T + \frac{\partial c_k}{\partial x}\frac{\partial x}{\partial p} \Sigma_p   \frac{\partial x}{\partial p}^T\frac{\partial c_k}{\partial x}^T, 
+
+All gradients are calculated with `k_aug <https://github.com/dthierry/k_aug>`_  [1]. More specifically, :math:`\frac{\partial f}{\partial p}, \frac{\partial f}{\partial x}, \frac{\partial c_1}{\partial p}, \frac{\partial c_1}{\partial x}, \ldots, \frac{\partial c_k}{\partial p}, \frac{\partial c_k}{\partial x}`, are computed via automatic differentiation whereas :math:`\frac{\partial x}{\partial p}` are computed via nonlinear programming sensitivity theory.
+
+The covariance matrix :math:`\Sigma_p` is either user supplied or obtained via regression (with ParmEst). 
+
+**Dependencies**
+
+`k_aug <https://github.com/dthierry/k_aug>`_ [1] is required to use uncertainty_propagation module. If you have the IDAES framework installed, you can get k_aug with the IDAES extensions by running the following command:
+
+.. code-block:: python
+
+   idaes get-extensions
+
+Basic Usage
+------------
+The ``uncertainty_propagation``'s main function is **quantify_propagate_uncertainty**. The quantify_propagate_uncertainty python function can have eight arguments and returns a namedtuple output. 
+
+The following example shows a usage of **quantify_propagate_uncertainty** with a Rooney and Biegler model [2].
+
+.. code:: python
+
+   # Required imports
+   >>> from idaes.apps.uncertainty_propagation.uncertainties import quantify_propagate_uncertainty
+   >>> import pyomo.environ as *
+   >>> import pandas as pd
+
+   # Define rooney_biegler_model
+   >>> def rooney_biegler_model(data):
+           model = ConcreteModel()
+           model.asymptote = Var(initialize = 15)
+           model.rate_constant = Var(initialize = 0.5)
+           
+           def response_rule(m, h):
+               expr = m.asymptote * (1 - exp(-m.rate_constant * h))
+               return expr
+           model.response_function = Expression(data.hour, rule = response_rule)
+    
+           def SSE_rule(m):
```

Comment:
Please see if parmest will still work as expected if you remove `SSE_rule`. I think it is redundant to define it twice.

Reply:
It works.